src/context/userDataContext.js

```diff
@@ -0,0 +1,29 @@
+import { createContext, useReducer, useEffect, useContext } from "react";
+import { userDataReducer } from "../reducer/userDataReducer";
```

Comment:
You can do the same single file export for reducers, contexts, pages and components files as well. Just make index.js files in the respective folders and export from there